### PR TITLE
implement #find with array of ids

### PR DESCRIPTION
**Pull request by jdahlke**

### CODE REVIEW

File: lib/crm/helpers/finders.rb [hunk 1]
```diff
@@ -5,11 +5,20 @@ def self.included(base)
...
+        when String
+          crm_object = crm_class.find(id_or_ids)
+          return nil if crm_object.blank?
 
-        new(crm_object.attributes)
+          new(crm_object.attributes)
+        when Array
+          crm_objects = Crm.find(id_or_ids).to_a
+          rejected_crm_objects = crm_objects.reject { |crm_object| crm_object.is_a?(crm_class) }
+          raise Errors::ResourceNotFound.new("Not of type #{crm_class.name}", rejected_crm_objects.map(&:id)) if rejected_crm_objects.present?
```
Comment (Skudo) [inline lib/crm/helpers/finders.rb]: I'd probably just stick with the standard `Errors::ResourceNotFound` message.

```ruby
raise Errors::ResourceNotFound.new('Items could not be found.', rejected_crm_objects.map(&:id)) if rejected_crm_objects.present?
```

Reason being that's what you're getting for doing something like `Crm::Account.find(contact_id)`, too. Well, you're getting something similar, that is...

**Comment by Skudo** [comment]
`ActiveRecord::FinderMethods#find` accepts these:

```ruby
Something.find(id)
#=> object
Something.find([id])
#=> [object]
Something.find(id, another_id)
#=> [object, another_object]
Something.find([id, another_id])
#=> [object, another_object]
```

We might want to consider the splat case, too. Maybe something like this:

```ruby
def find(*ids)
  expects_array = ids.first.is_a?(Array)
  ids.flatten!

  case ids.size
  when 0
    raise Errors::BaseError('Could not find item without ID.')
    # Maybe create a custom error class for this one...
    # Maybe just raise ArgumentError instead...
  when 1
    crm_object = find_one(ids.first)
    expects_array ? [crm_object] : crm_object
  else
    find_many(ids)
  end
end

protected

def find_one(id)
  crm_object = crm_class.find(id)
  new(crm_object.attributes)
end

def find_many(array_of_ids)
  crm_objects = Crm.find(array_of_ids).select do |crm_object|
    crm_object.is_a?(crm_class)
  end
  unknown_ids = array_of_ids - crm_objects.map(&:id)
  raise Errors::ResourceNotFound('Items could not be found.', unknown_ids) if unknown_ids.present?

  crm_objects.map { |crm_object| new(crm_object.attributes) }
end

**Comment by Skudo** [comment]
Oh, and specs!

**Comment by Skudo** [comment]
In either case, specs for the "some ids could not be found" case are missing, as far as I can see.

**Comment by jdahlke** [comment]
@Skudo added your feedback

**Pull request merged**